src/components/ErrorBoundary/ErrorBoundary.tsx

```diff
@@ -45,37 +43,38 @@ export class ErrorBoundary extends Component<ErrorBoundaryProps, ErrorBoundarySt
     window.removeEventListener("unhandledrejection", this.onUnhandledRejection);
   }
 
+  /**
+   * Global promise unhandled promise rejection to trigger error boundary
+   */
   onUnhandledRejection = (event: PromiseRejectionEvent): void => {
-    const { onError } = this.props;
     event.preventDefault();
     event.promise.catch(async (error) => {
-      stack(error);
-      this.setState(ErrorBoundary.getDerivedStateFromError(error), () => {
-        if (onError) onError(error);
-      });
+      this.setState(ErrorBoundary.getDerivedStateFromError(error));
     });
   };
 
+  /**
+   * Attempts to recover from error.
+   */
   recover = (): void => {
     const { onRecover } = this.props;
-    this.setState(
-      {
-        hasError: false,
-        error: undefined,
-      },
-      () => {
-        if (onRecover) onRecover();
-      }
-    );
+    onRecover();
```

Comment:
this part i didn't really und. the prev code:

```
this.setState(
      {
        hasError: false,
        error: undefined,
      },
      () => {
        if (onRecover) onRecover();
      }
    );
```

the above ensures that `onRecover` gets called immediately after `setState` is done

so what you meant is the above is safer as recover will be "tied" to when setState (error reset), in case if in any event react did not trigger re-render (due to similar error, retry)

^ correct my understanding if wrong